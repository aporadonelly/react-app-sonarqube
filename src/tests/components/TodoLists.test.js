import React from "react";
import "jest-enzyme";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TodoLists from "../../components/TodoLists";
configure({ adapter: new Adapter() });

describe("Todo", () => {
  it("should test TodoLists component with default state of empty array", () => {
    const wrapper = shallow(<TodoLists todos={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should test TodoLists component with list of todos", () => {
    const wrapper = shallow(
      <TodoLists todos={["Read Newspaper", "Publish Article"]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
