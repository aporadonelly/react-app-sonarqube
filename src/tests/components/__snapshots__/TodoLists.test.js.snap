// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should test TodoLists component with default state of empty array 1`] = `
<div
  className="todo-list"
>
  <p>
    No todos found. Please add some.
  </p>
</div>
`;

exports[`Todo should test TodoLists component with list of todos 1`] = `
<div
  className="todo-list"
>
  <TodoItem
    key="0"
    todo="Read Newspaper"
  />
  <TodoItem
    key="1"
    todo="Publish Article"
  />
</div>
`;

exports[`should test TodoLists component with default state of empty array 1`] = `
<div
  className="todo-list"
>
  <p>
    No todos found. Please add some.
  </p>
</div>
`;

exports[`should test TodoLists component with list of todos 1`] = `
<div
  className="todo-list"
>
  <TodoItem
    key="0"
    todo="Read Newspaper"
  />
  <TodoItem
    key="1"
    todo="Publish Article"
  />
</div>
`;
